### PR TITLE
Add cask joplin with version 0.10.31

### PR DESCRIPTION
**Pull request by OranShuster**
After making all changes to the cask:

- [X] `brew cask audit --download {{cask_file}}` is error-free.
- [X] `brew cask style --fix {{cask_file}}` reports no offenses.
- [X] The commit message includes the cask’s name and version.

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [X] `brew cask install {{cask_file}}` worked successfully.
- [X] `brew cask uninstall {{cask_file}}` worked successfully.
- [X] Checked there are no [open pull requests] for the same cask.
- [X] Checked the cask was not already refused in [closed issues].
- [X] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-cask/pulls
[closed issues]: https://github.com/caskroom/homebrew-cask/issues?q=is%3Aissue+is%3Aclosed
[the correct repo]: https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

**Comment by OranShuster** [comment]
Didn't do zap/uninstall since i don't really know how to find all the places it writes to

File: Casks/joplin.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+cask 'joplin' do
+  version '0.10.28'
+  sha256 '71a9dc3fb9041b803982458246ee14bf62e595e83738ead48c7b86e38bdf339d'
+
+  url "https://github.com/laurent22/joplin/releases/download/v#{version}/Joplin-#{version}.dmg"
```
Comment (commitay) [inline Casks/joplin.rb]: Please add an [`appcast`](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/appcast.md)

Example:
https://github.com/caskroom/homebrew-cask/blob/0480e1f119f2742586f35b65a4efbfba670757b8/Casks/autodmg.rb#L5-L8

File: Casks/joplin.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  name 'joplin'
```
Comment (commitay) [inline Casks/joplin.rb]: Joplin

File: Casks/joplin.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+  homepage 'https://github.com/laurent22/joplin'
+
+  app "Joplin.app"
```
Comment (commitay) [inline Casks/joplin.rb]: `app 'Joplin.app'`

**Review by commitay — DISMISSED** [review]

**Comment by commitay** [comment]
Please don't squash commits, it makes it harder to see it changes have been made.

**Comment by OranShuster** [comment]
Sorry, this should be the final commit

**Comment by commitay** [comment]
I've updated the homepage and added a zap stanza.

**Comment by commitay** [comment]
I'll bump this to the latest version immediately after this is merged.

**Comment by commitay** [comment]
Finally get through the CI backlog after new upstream releases and ... 
```
audit for joplin: warning
 - possible duplicate, cask token conflicts with Homebrew core formula: https://github.com/Homebrew/homebrew-core/blob/master/Formula/joplin.rb
```
Formula was merged after this first failed `audit`

Merging this as [formula overlap](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md#with-a-formula) rule does not apply, this is a proper `app`, not a cli wrapper.

**Review by commitay — APPROVED** [review]

**Pull request merged**